src/xdsl/error_format.py

```diff
@@ -0,0 +1,134 @@
+import os
+import sys
+from types import TracebackType
+import traceback
+import inspect
```

Comment:
I would advice using explicit import instead of importing whole package. Btw, can also use isort to sort the imprort.